File: moban_java/src/main/java/org/jdk/project/controller/ProfileController.java
package org.jdk.project.controller;

import java.security.Principal;
import lombok.RequiredArgsConstructor;
import org.jdk.project.dto.user.UpdateProfileRequest;
import org.jdk.project.dto.user.UserDto;
import org.jdk.project.service.UserService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/me")
@RequiredArgsConstructor
public class ProfileController {

  private final UserService userService;

  @GetMapping
  public UserDto getProfile(Principal principal) {
    return userService.getProfile(principal.getName());
  }

  @PutMapping
  public UserDto updateProfile(
      Principal principal, @RequestBody UpdateProfileRequest request) {
    return userService.updateProfile(principal.getName(), request);
  }

  @PutMapping("/password")
  public void updatePassword(
      Principal principal,
      @RequestParam String oldPassword,
      @RequestParam String newPassword) {
    userService.updatePassword(principal.getName(), oldPassword, newPassword);
  }

  @PutMapping("/avatar")
  public UserDto updateAvatar(
      Principal principal, @RequestParam String avatarUrl) {
    return userService.updateAvatar(principal.getName(), avatarUrl);
  }
}

File: backend/src/main/kotlin/org/jdk/project/controller/ProfileController.kt
package org.jdk.project.controller

import java.security.Principal
import org.jdk.project.dto.user.UpdateProfileRequest
import org.jdk.project.dto.user.UserDto
import org.jdk.project.service.UserService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/me")
class ProfileController(
    private val userService: UserService,
) {
    @GetMapping
    fun getProfile(principal: Principal): UserDto = userService.getProfile(principal.name)

    @PutMapping
    fun updateProfile(principal: Principal, @RequestBody request: UpdateProfileRequest): UserDto =
        userService.updateProfile(principal.name, request)

    @PutMapping("/password")
    fun updatePassword(
        principal: Principal,
        @RequestParam oldPassword: String,
        @RequestParam newPassword: String,
    ) {
        userService.updatePassword(principal.name, oldPassword, newPassword)
    }

    @PutMapping("/avatar")
    fun updateAvatar(principal: Principal, @RequestParam avatarUrl: String): UserDto =
        userService.updateAvatar(principal.name, avatarUrl)
}
